Ingest flows

vgeorge opened the pull request:
Closes https://github.com/earth-genome/foodtwin-global-app/issues/26

Opening this for visibility, this aims to implement the ingest workflow for the food flows.

We scaffolded the ingest script in #8 and #12, which currently covers the following datasets:

- Admin area centroids
- Admin area limits
- Maritime nodes
- Maritime edges

Considering that the flow dataset depends on others (land/sea edges, food groups), I suggest we work on the files sequentially, ensuring each dataset can be ingested correctly before moving to the next one.

At the moment, we are working on ingesting the edges for land. In the v3 data folder, there is the file `Geometry_Landmapping_hexcode_complete.csv`, which includes the geometries but not the references nodes (e.g., `rail1870326_SDN-rail1870327_SDN`).

Next steps:

- [ ] Add land nodes file to v3 folder
- [ ] Land nodes ingest step
- [ ] Land edges ingest step (already scaffolded, but needs the nodes to be completed)
- [ ] Food groups ingest (file `UniqueFG1_FG2.csv` seems to include the required data)
- [ ] Flows ingest

cc @cameronkruse  @faustoperez  @oliverroick @wrynearson

kushankb commented:
I have added IWW and rail node files to the folder [here](https://drive.google.com/drive/folders/1JYyYQ5XkwCAMSy3-FHanMbVR-ex38R2k?usp=drive_link) (I'll let you move them to v3), which include latitude and longitude information, as well as details indicating whether each node is an IWW port or rail station. However, I am currently missing a matching key to link these nodes with their official names (e.g., Port of Shanghai). I'm waiting to hear back from Jasper about obtaining the matching key. Let me know if these files help you continue the ingestion process.
 @vgeorge

vgeorge commented:
@kushankb thanks for providing the files, I've moved them to the v3 folder and updated the seed script. We can add the names later once you have they ready. The last step to complete the edge ingest is to add missing road nodes, can you please point to them? I added a file to `v3` folder called `unmatched_land_edges.csv` that lists the missing nodes.

@oliverroick I wont be adding changes to this until I'm back from holidays. To run this locally, you must download the `v3` files and expand the compacted geometry file inside it.

@wrynearson @faustoperez below is a preview from the current data after is was ingested to the database. The horizontal lines are probably from features that cross the International Day Line. Unfortunately I haven't found a way to hide the on QGIS, I hope it is still helpful:

<img width="1248" alt="Screenshot 2024-08-14 at 13 51 17" src="https://github.com/user-attachments/assets/a68bb9dd-cdd1-44de-a90b-68d4656fd46f">

kushankb commented:
Hey @vgeorge, I've added the road nodes data to the same folder [here](https://drive.google.com/drive/folders/1JYyYQ5XkwCAMSy3-FHanMbVR-ex38R2k). I am looking into the missing road edges now.

wrynearson commented:
Thanks @kushankb! Maybe @oliverroick can answer that question while @vgeorge is away? He'll be back on the 26th.
